### ISSUE TITLE
When the model isn't sure what comes next, don't return anything?

### ISSUE THREAD

**Issue by ronnyli** [minimaxir/gpt-2-simple#126]
**TL;DR: Is there a function like `gpt2.generate` that returns logits or some measure of how uncertain the model is about the sampled text?**

I have a dataset of Reddit comments and their associated summaries (like the [TL;DR dataset](https://www.aclweb.org/anthology/W17-4508/)). The format is `<|startoftext|>comment_text<|TLDR|>tldr_text<|endoftext|>`. When I use `gpt2.generate` I set the prefix to be everything up to `<|TLDR|>` so the model will generate a summary. This works pretty well when the `comment_text` is long enough to give the model context for the summary.

However, sometimes the model generates random summaries across a whole range of topics that have nothing to do with the original comment text. This is usually because the comment is too short to provide context. I'd like to be able to detect when this is the case. Any suggestions would be greatly appreciated :)

**Comment by minimaxir** [comment]
Unfortunately, this isn't possible to implement reliably.

The way to calculate "uncertainty" would be to take the maximum probability of the next token; if it's very low (e.g. < 0.1) we could say the model is uncertain. If the probability is high (> 0.5) we could say the model is certain.

But when the model goes on a tangent, it can still be _certain_ of the next token in its tangential behavior.

**Comment by ronnyli** [comment]
Thanks for the reply. Since I'm creating a summary of the text I can probably filter out the ones that have a low ROUGE score or something like that.
